### PR TITLE
chore: example of how to run eslint under test with bazel

### PR DESCRIPTION
**Pull request by Aghassi**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: Adds an example to the example folder

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: examples/eslint/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    out = ".eslintignore",
+    content = [
+        "lint_loader.js",
+        "lint_require_patch.js",
+        "_lint.module_mappings.json",
+        "lint.sh",
+    ],
+)
+
+write_file(
```
Comment (alexeagle) [inline examples/eslint/BUILD.bazel:24]: remove this now that we have `chdir` attribute

File: examples/eslint/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,58 @@
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@npm//eslint:index.bzl", "eslint_test")
+
+# Make file available to lint call below
+exports_files(
+    [
+        ".eslintrc.js",
```
Comment (alexeagle) [inline examples/eslint/BUILD.bazel:7]: this shouldn't be needed since it's in the same package

File: examples/eslint/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    ],
+    data = [
+        # This will automatically be read by eslint
+        ":eslintignore",
+        ":write_eslint_chdir_script",
+        ".eslintrc.js",
+        # Needed for Bazel coverage
+        "@npm//c8",
+        # ESLint config and dependencies
+        "@npm//eslint-config-airbnb",
```
Comment (alexeagle) [inline examples/eslint/BUILD.bazel:50]: I understand why `airbnb` is here but why the other dependencies? doesn't eslint already depend on them?

Comment (Aghassi) [reply examples/eslint/BUILD.bazel:50]: they are peer dependencies of the airbnb config

**Comment by Aghassi** [comment]
@alexeagle this will fail until a new `3.x.x` release is cut with the `chdir` change in it I think.

**Comment by alexeagle** [comment]
We do a trick to replace the dependencies with HEAD when running the examples and e2e tests, so it ought to work.

**Comment by alexeagle** [comment]
@googlebot I consent.

**Pull request closed without merging**